modules/document_repository/php/document_repository.class.inc

```diff
@@ -37,208 +36,92 @@ class Document_Repository extends \NDB_Menu_Filter
      */
     function _hasAccess(\User $user) : bool
     {
-        return $user->hasPermission('document_repository_view')
-               || $user->hasPermission('document_repository_delete');
-
+        return $user->hasAnyPermission(
+            array(
+             'document_repository_view',
+             'document_repository_delete',
+            )
+        );
     }
-
     /**
      * Function _setupVariables
      *
      * @return void
      */
-    function _setupVariables()
+    function setup()
     {
-
-        $user =& \User::singleton();
-
-        // create the centerID map
-        $db      =& \Database::singleton();
-        $pscRows = $db->pselect("SELECT CenterID, Name FROM psc", array());
-        foreach ($pscRows AS $row) {
-            $this->centerIDMap[$row['CenterID']] = $row['Name'];
+        parent::setup();
+        $stdin     = file_get_contents('php://input');
+        $req       = json_decode($stdin, true);
```

Comment:
I don't understand this.. why is setup trying to read from stdin on the main page?